SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorCloudLauncherSettings.h

```diff
@@ -0,0 +1,136 @@
+class SPATIALGDKEDITOR_API USpatialGDKEditorCloudLauncherSettings : public UObject
+{
+	GENERATED_BODY()
+
+public:
+	USpatialGDKEditorCloudLauncherSettings(const FObjectInitializer& ObjectInitializer);
+
+private:
+	/** Path to the directory containing the SpatialOS-related files. */
+	UPROPERTY(EditAnywhere, config, Category = "General", meta = (ConfigRestartRequired = false, DisplayName = "SpatialOS directory"))
+		FDirectoryPath SpatialOSDirectory;
+
+private:
+	UPROPERTY(config)
+	FString ProjectName;
+
+	UPROPERTY(config)
+	FString PrimaryDeploymentName;
+
+	UPROPERTY(config)
+	FString AssemblyName;
+
+	UPROPERTY(config)
+	FFilePath PrimaryLaunchConfigPath;
+
+	UPROPERTY(config)
+	FFilePath SnapshotPath;
+
+	UPROPERTY(config)
+	FString SimulatedPlayerDeploymentName;
+
+	UPROPERTY(config)
+	FFilePath SimulatedPlayerLaunchConfigPath;
+
+	UPROPERTY(config)
+	uint32 NumberOfSimulatedPlayers;
+
+	bool bPrimaryDeploymentNameIsValid;
+	bool bAssemblyNameIsValid;
+	bool bProjectNameIsValid;
+
+	UPROPERTY(config)
+	bool SimulatedPlayersIsEnabled;
+
+	void ValidateAssemblyName();
+	void ValidateProjectName();
+	void ValidateDeploymentName();
+
+public:
+	FString GetProjectNameFromSpatial();
+
+	void SetPrimaryDeploymentName(const FString & Name);
+	FORCEINLINE FString GetPrimaryDeploymentName() const
+	{
+		return PrimaryDeploymentName;
+	}
+
+	void SetAssemblyName(const FString & Name);
+	FORCEINLINE FString GetAssemblyName() const
+	{
+		return AssemblyName;
+	}
+
+	void SetProjectName(const FString & Name);
+	FORCEINLINE FString GetProjectName() const
+	{
+		return ProjectName;
+	}
+
+	void SetPrimaryLaunchConfigPath(const FString & Path);
+	FORCEINLINE FString GetPrimaryLanchConfigPath() const
+	{
+		return PrimaryLaunchConfigPath.FilePath;
+	}
+
+	void SetSnapshotPath(const FString & Path);
+	FORCEINLINE FString GetSnapshotPath() const
+	{
+		return SnapshotPath.FilePath;
+	}
+
+	void SetSimulatedPlayersEnabledState(bool IsEnabled);
+	FORCEINLINE bool IsSimulatedPlayersEnabled() const
+	{
+		return SimulatedPlayersIsEnabled;
+	}
+
+	void SetSimulatedPlayerDeploymentName(const FString & Name);
+	FORCEINLINE FString GetSimulatedPlayerDeploymentName() const
+	{
+		return SimulatedPlayerDeploymentName;
+	}
+
+	void SetSimulatedPlayerLaunchConfigPath(const FString & Path);
+	FORCEINLINE FString GetSimulatedPlayerLaunchConfigPath() const
+	{
+		return SimulatedPlayerLaunchConfigPath.FilePath;
+	}
+
+	void SetNumberOfSimulatedPlayers(uint32 Number);
+	FORCEINLINE uint32 GetNumberOfSimulatedPlayer() const
+	{
+		return NumberOfSimulatedPlayers;
+	}
+
+	FORCEINLINE bool IsProjectNameValid() const
+	{
+		return bProjectNameIsValid;
+	}
+
+	FORCEINLINE bool IsPrimaryDeploymentNameValid() const
+	{
+		return bPrimaryDeploymentNameIsValid;
+	}
+
+	FORCEINLINE bool IsAssemblyNameValid() const
+	{
+		return bAssemblyNameIsValid;
+	}
+
+	bool IsDeploymentConfigurationValidSinceLastCheck() const;
```

Comment:
Can this be a private function?

Reply:
The intention is for this function to be used when some UI logic depends on the deployment configuration being valid without executing the validation function explicitly.

Reply:
Do we have any code doing that at the moment?

Reply:
Discussed offline - will go with the approach of validating the configuration state only when the button is pressed. 🎉